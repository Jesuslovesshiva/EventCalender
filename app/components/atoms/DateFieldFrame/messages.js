import {defineMessages} from 'react-intl'

const scope = 'atoms.date-field-frame'

export default defineMessages({
  toggle: {
    id: `${scope}.toggle`,
    defaultMessage: 'Kalendar ein-/ausschalten',
    description: 'aria label toggle calendar button'
  }
})
