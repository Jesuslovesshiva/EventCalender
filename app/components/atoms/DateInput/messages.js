import {defineMessages} from 'react-intl'

const scope = 'atoms.date-picker'

export default defineMessages({
  ddmmyyyy: {
    id: `${scope}.date`,
    defaultMessage: 'TT.MM.JJJJ',
    description: 'placeholder for format dd.MM.yyyy'
  }
})
