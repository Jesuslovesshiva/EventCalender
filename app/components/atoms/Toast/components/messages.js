import {defineMessages} from 'react-intl'

const scope = 'atoms.toast'

export default defineMessages({
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Info'
  },
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Error or Warning'
  },
  alert: {
    id: `${scope}.alert`,
    defaultMessage: 'alert'
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'Action'
  }
})
