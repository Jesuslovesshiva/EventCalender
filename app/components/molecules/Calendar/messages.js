import {defineMessages} from 'react-intl'

const scope = 'molecules.calendar'

export default defineMessages({
  selectMonth: {
    id: `${scope}.selectMonth`,
    defaultMessage: 'Monat auswählen'
  },
  selectYear: {
    id: `${scope}.selectYear`,
    defaultMessage: 'Jahr auswählen'
  },
  nextMonth: {
    id: `${scope}.nextMonth`,
    defaultMessage: 'nächstes Monat',
    description: 'aria label nextMonth button'
  },
  prevMonth: {
    id: `${scope}.prevMonth`,
    defaultMessage: 'vorheriges Monat',
    description: 'aria label prevMonth button'
  }
})
