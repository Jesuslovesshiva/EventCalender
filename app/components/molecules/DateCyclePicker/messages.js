import {defineMessages} from 'react-intl'

const scope = 'molecules.date-cycle-picker'

export default defineMessages({
  from: {
    id: `${scope}.from`,
    defaultMessage: 'von',
    description: 'aria label from input'
  },
  to: {
    id: `${scope}.to`,
    defaultMessage: 'bis',
    description: 'aria label to input'
  },
  next: {
    id: `${scope}.next`,
    defaultMessage: 'n√§chste Woche',
    description: 'aria label next button'
  },
  prev: {
    id: `${scope}.prev`,
    defaultMessage: 'vorherige Woche',
    description: 'aria label prev button'
  }
})
