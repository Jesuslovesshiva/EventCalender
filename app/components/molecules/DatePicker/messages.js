import {defineMessages} from 'react-intl'

const scope = 'molecules.date-picker'

export default defineMessages({
  date: {
    id: `${scope}.date`,
    defaultMessage: 'Datum',
    description: 'aria label date input'
  }
})
