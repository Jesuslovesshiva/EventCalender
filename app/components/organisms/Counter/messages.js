import {defineMessages} from 'react-intl'

const scope = 'organisms.counter'

export default defineMessages({
  count: {
    id: `${scope}.count`,
    defaultMessage: 'Zähler: {count}',
    description: 'count output'
  },
  reset: {
    id: `${scope}.reset`,
    defaultMessage: 'Zurücksetzen',
    description: 'reset cta'
  }
})
