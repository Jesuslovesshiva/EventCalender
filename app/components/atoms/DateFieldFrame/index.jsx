import classNames from 'classnames'
import noop from 'lodash/noop'
import PropTypes from 'prop-types'
import {forwardRef, useCallback} from 'react'
import {useIntl} from 'react-intl'

import ArrowDropDownIcon from '@/components/atoms/Icon/ArrowDropDown'
import DateRangeIcon from '@/components/atoms/Icon/DateRange'

import messages from './messages'

const DateFieldFrame = forwardRef((props, ref) => {
  const {formatMessage} = useIntl()

  const onToggle = useCallback(
    event => {
      event.stopPropagation()
      props.onToggle()
    },
    [props.onToggle]
  )

  return (
    <div
      ref={ref}
      className={classNames(
        'border border-gray-medium divide-x divide-gray-medium rounded inline-flex transition cursor-pointer',
        'hover:border-white hover:divide-white hover:bg-silver hover:shadow-md specific:divider-gray-light',
        {
          'specific:border-white specific:divide-gray-light specific:bg-white':
            props.isOpen,
          'specific:shadow specific:rounded-b-none': props.isOpen
        }
      )}
    >
      {props.previousButton}

      <div
        className={classNames('flex items-center py-1 transition', {
          'pl-1.5': Boolean(props.previousButton),
          'pl-3': !props.previousButton,
          'pr-1.5': Boolean(props.nextButton),
          'pr-2': !props.nextButton
        })}
        onClick={props.onCenterClick}
      >
        <DateRangeIcon className="text-lg mr-1" />

        {props.children}

        <button
          aria-label={formatMessage(messages.toggle)}
          className="transition text-gray-medium hover:text-gray-dark"
          onClick={onToggle}
        >
          <ArrowDropDownIcon
            className={classNames('text-2xl', {'-rotate-180': props.isOpen})}
          />
        </button>
      </div>

      {props.nextButton}
    </div>
  )
})

DateFieldFrame.displayName = 'DateFieldFrame'

DateFieldFrame.propTypes = {
  children: PropTypes.node,
  isOpen: PropTypes.bool,
  nextButton: PropTypes.node,
  onCenterClick: PropTypes.func,
  onToggle: PropTypes.func,
  previousButton: PropTypes.node
}

DateFieldFrame.defaultProps = {
  children: null,
  isOpen: false,
  nextButton: null,
  onCenterClick: noop,
  onToggle: noop,
  previousButton: null
}

export default DateFieldFrame
