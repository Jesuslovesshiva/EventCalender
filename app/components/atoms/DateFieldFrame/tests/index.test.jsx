import {fireEvent} from '@testing-library/react'

import render from '@/utils/test-utils/custom-render'

import DateFieldFrame from '../index'
import messages from '../messages'

const renderComponent = (props, renderFn = render) =>
  renderFn(<DateFieldFrame {...props} />)

describe('DateFieldFrame', () => {
  it('should match the snapshot', () => {
    const {container, rerender} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
    renderComponent({isOpen: true}, rerender)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render previousButton and nextButton props', () => {
    const {getByText} = renderComponent({
      nextButton: <div>nextButton</div>,
      previousButton: <div>previousButton</div>
    })
    expect(getByText('previousButton')).toBeInTheDocument()
    expect(getByText('nextButton')).toBeInTheDocument()
  })

  it('should call onToggle prop on toggle click', () => {
    const onToggle = jest.fn()
    const {getByLabelText} = renderComponent({onToggle})
    const toggle = getByLabelText(messages.toggle.defaultMessage)
    fireEvent.click(toggle)
    expect(onToggle).toHaveBeenCalled()
  })

  it('should call onToggle prop on toggle click', () => {
    const onCenterClick = jest.fn()
    const {container} = renderComponent({onCenterClick})
    fireEvent.click(container.firstChild.firstChild)
    expect(onCenterClick).toHaveBeenCalled()
  })
})
