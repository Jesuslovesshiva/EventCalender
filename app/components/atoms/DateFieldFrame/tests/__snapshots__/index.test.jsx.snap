// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFieldFrame should match the snapshot 1`] = `
<div
  class="border border-gray-medium divide-x divide-gray-medium rounded inline-flex transition cursor-pointer hover:border-white hover:divide-white hover:bg-silver hover:shadow-md specific:divider-gray-light"
>
  <div
    class="flex items-center py-1 transition pl-3 pr-2"
  >
    <div
      class="fill-current w-[1em] h-[1em] inline-block text-lg mr-1"
      data-icon="date_range"
    />
    <button
      aria-label="Kalendar ein-/ausschalten"
      class="transition text-gray-medium hover:text-gray-dark"
    >
      <div
        class="fill-current w-[1em] h-[1em] inline-block text-2xl"
        data-icon="arrow_drop_down"
      />
    </button>
  </div>
</div>
`;

exports[`DateFieldFrame should match the snapshot 2`] = `
<div
  class="border border-gray-medium divide-x divide-gray-medium rounded inline-flex transition cursor-pointer hover:border-white hover:divide-white hover:bg-silver hover:shadow-md specific:divider-gray-light specific:border-white specific:divide-gray-light specific:bg-white specific:shadow specific:rounded-b-none"
>
  <div
    class="flex items-center py-1 transition pl-3 pr-2"
  >
    <div
      class="fill-current w-[1em] h-[1em] inline-block text-lg mr-1"
      data-icon="date_range"
    />
    <button
      aria-label="Kalendar ein-/ausschalten"
      class="transition text-gray-medium hover:text-gray-dark"
    >
      <div
        class="fill-current w-[1em] h-[1em] inline-block text-2xl -rotate-180"
        data-icon="arrow_drop_down"
      />
    </button>
  </div>
</div>
`;
